packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2396,6 +2396,7 @@ export default (superClass: Class<Parser>): Class<Parser> =>
       const callParseClassMemberWithIsStatic = () => {
         const isStatic = !!member.static;
         if (isStatic && this.eat(tt.braceL)) {
+          delete member.static;
```

Comment:
Deleting a property should be generally avoided (https://stackoverflow.com/a/44008788/1490357), especially when it is executed for normal execution path: static blocks. I think we can check if a `tt.braceL` is upcoming before `member.static` is assigned in `tsParseModifiers`.

Reply:
It was difficult to avoid to attach `static` in `tsParseModifiers` without failing existing tests. So what do you think about to separate attaching modifier properties and parsing ( d5077ff )?